jscomp/runtime/js_typed_array.cppo.ml

```diff
@@ -180,100 +69,84 @@ end
   (* There's also an overload for typed arrays, but don't know how to model that without subtyping *)\
   \
   (* Array interface(-ish) *)\
   external length : t -> int = "length" [@@mel.get]\
   \
   (* Mutator functions *)\
-  external copyWithin : to_:int -> t = "copyWithin" [@@mel.send.pipe: t]\
-  external copyWithinFrom : to_:int -> from:int -> t = "copyWithin" [@@mel.send.pipe: t]\
-  external copyWithinFromRange : to_:int -> start:int -> end_:int -> t = "copyWithin" [@@mel.send.pipe: t]\
+  external copyWithin : t -> to_:int -> ?start:int -> ?end_:int -> unit -> t = "copyWithin" [@@mel.send]\
   \
-  external fillInPlace : elt -> t = "fill" [@@mel.send.pipe: t]\
-  external fillFromInPlace : elt -> from:int -> t = "fill" [@@mel.send.pipe: t]\
-  external fillRangeInPlace : elt -> start:int -> end_:int -> t = "fill" [@@mel.send.pipe: t]\
+  external fill : t -> elt -> ?start:int -> ?end_:int -> unit -> t = "fill" [@@mel.send]\
   \
-  external reverseInPlace : t = "reverse" [@@mel.send.pipe: t]\
+  external reverseInPlace : t -> t = "reverse" [@@mel.send]\
   \
-  external sortInPlace : t = "sort" [@@mel.send.pipe: t]\
-  external sortInPlaceWith : (elt -> elt -> int [@u]) -> t = "sort" [@@mel.send.pipe: t]\
+  external sortInPlace : t -> t = "sort" [@@mel.send]\
+  external sortInPlaceWith : t -> f:(elt -> elt -> int [@u]) -> t = "sort" [@@mel.send]\
   \
   (* Accessor functions *)\
-  external includes : elt -> bool = "includes" [@@mel.send.pipe: t] (* ES2016 *)\
+  external includes : t -> value:elt -> bool = "includes" [@@mel.send] (* ES2016 *)\
   \
-  external indexOf : elt  -> int = "indexOf" [@@mel.send.pipe: t]\
-  external indexOfFrom : elt -> from:int -> int = "indexOf" [@@mel.send.pipe: t]\
+  external indexOf : t -> value:elt -> ?start:int -> unit -> int = "indexOf" [@@mel.send]\
   \
-  external join : string = "join" [@@mel.send.pipe: t]\
-  external joinWith : string -> string = "join" [@@mel.send.pipe: t]\
+  external join : t -> ?sep:string -> unit -> string = "join" [@@mel.send]\
   \
-  external lastIndexOf : elt -> int = "lastIndexOf" [@@mel.send.pipe: t]\
-  external lastIndexOfFrom : elt -> from:int -> int = "lastIndexOf" [@@mel.send.pipe: t]\
+  external lastIndexOf : t -> value:elt -> int = "lastIndexOf" [@@mel.send]\
+  external lastIndexOfFrom : t -> value:elt -> from:int -> int = "lastIndexOf" [@@mel.send]\
   \
-  external slice : start:int -> end_:int -> t = "slice" [@@mel.send.pipe: t]\
+  external slice : t -> ?start:int -> ?end_:int -> unit -> t = "slice" [@@mel.send]\
   (** [start] is inclusive, [end_] exclusive *)\
   \
-  external copy : t = "slice" [@@mel.send.pipe: t]\
-  external sliceFrom : int -> t = "slice" [@@mel.send.pipe: t]\
+  external copy : t -> t = "slice" [@@mel.send]\
   \
-  external subarray : start:int -> end_:int -> t = "subarray" [@@mel.send.pipe: t]\
+  external subarray : t -> start:int -> ?end_:int -> unit -> t = "subarray" [@@mel.send]\
```

Comment:
I think `start` is optional too: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/subarray